ops/storage-ddls/cassandra-setup.cql

```diff
@@ -79,87 +76,126 @@ CREATE TYPE features (
+    externalsourceid text,
+    topicconjunctionid uuid,
     insertion_time timestamp,
     computedfeatures frozen<features>,
-    PRIMARY KEY ((placeid), periodstartdate, periodenddate, pipeline, sourceid, topic)
+    PRIMARY KEY ((period, periodtype, pipelinekey, topicconjunctionid, placeid), externalsourceid, periodstartdate, periodenddate)
 );
 
-CREATE INDEX ON computedplaces (period);
-CREATE INDEX ON computedplaces (computedfeatures);
-CREATE INDEX ON computedplaces (periodtype);
-
-CREATE TABLE computedtopics (
+CREATE TABLE populartopics (
     periodstartdate timestamp,
     periodenddate timestamp,
     periodtype text,
     period text,
-    pipeline text,
-    sourceid text,
+    pipelinekey text,
+    externalsourceid text,
     topic text,
     topiclangcode text,
+    mentionCount int,
     insertion_time timestamp,
-    computedfeatures frozen<features>,
-    PRIMARY KEY ((topic), pipeline, periodstartdate, periodenddate, sourceid)
-);
+    PRIMARY KEY ((period, periodtype, pipelinekey, externalsourceid, topiclangcode), mentionCount, periodstartdate, periodenddate, topic)
+) WITH CLUSTERING ORDER BY (mentionCount DESC, periodstartdate DESC, periodenddate DESC, topic ASC);
+
+CREATE TABLE popularsources (
+    periodstartdate timestamp,
+    periodenddate timestamp,
+    periodtype text,
+    period text,
+    pipelinekey text,
+    externalsourceid text,
+    topicconjunctionid uuid,
+    mentionCount int,
+    insertion_time timestamp,
+    PRIMARY KEY ((period, periodtype, pipelinekey, topicconjunctionid), mentionCount, periodstartdate, periodenddate, externalsourceid)
+) WITH CLUSTERING ORDER BY (mentionCount DESC, periodstartdate DESC, periodenddate DESC, externalsourceid ASC);
 
-CREATE INDEX ON computedtopics (period);
-CREATE INDEX ON computedtopics (computedfeatures);
-CREATE INDEX ON computedtopics (periodtype);
+CREATE TABLE popularplaces (
+    periodstartdate timestamp,
+    periodenddate timestamp,
+    periodtype text,
+    period text,
+    pipelinekey text,
+    externalsourceid text,
+    placename text,
+    placeid int,
+    topicconjunctionid uuid,
+    mentionCount int,
+    insertion_time timestamp,
+    PRIMARY KEY ((period, periodtype, pipelinekey, topicconjunctionid, externalsourceid), mentionCount, periodstartdate, periodenddate, placename)
+) WITH CLUSTERING ORDER BY (mentionCount DESC, periodstartdate DESC, periodenddate DESC, placename ASC);
 
 CREATE TABLE events(
     id uuid,
     externalid text,
-    pipeline text,
+    pipelinekey text,
     title text,
     sourceurl text,
     detectedplaceids frozen<set<text>>,
-    sourceid text,
-    detectedkeywords frozen<set<text>>,
+    externalsourceid text,
+    topicconjunctionids set<text>,
     eventlangcode text,
     messagebody text,
     computedfeatures frozen<features>,
     insertion_time timestamp,
     event_time timestamp,
-    PRIMARY KEY (pipeline, externalid)
+    PRIMARY KEY ((pipelinekey, eventlangcode), externalid)
```

Comment:
Could you speak to the benefits of this primary key over just using the `id`?

If I understood some of our earlier conversations correctly, we want to avoid querying the events table because it'll be super large, so most of our queries will be against `computedXYZ` tables. One of the few cases where we actually need to query for particular event details is for the following resolver:

```js
/**
 * @param {{site: string, messageId: string}} args
 * @returns {Promise.<Feature>}
 */
function event(args, res)
```

If I understand correctly, this resolver is called when we go to an event details page (see screenshot below) (e.g. following a link from the newsfeed), so it would be easiest to be able to query the table just by id.

<img src="https://user-images.githubusercontent.com/1086421/28139758-4b41364c-670a-11e7-9bb3-bca555d6c1ca.png" width=400 />